### ISSUE TITLE
Replace `rhdf5` with `h5`

### ISSUE THREAD

**Issue by ashiklom** [PecanProject/pecan#1797]
## Description

The [`h5` package](https://cran.r-project.org/web/packages/h5/) provides a convenient, object-oriented interface to HDF5 files. Creating files is as easy as:

```r
library(h5)
hfile <- h5file("test.h5")
d1 <- hfile["group1/dataset1"]
d1[] <- rnorm(100)
h5attr(d1, "index") <- seq_len(100)
h5attr(d1, "id") <- paste0("a", seq_len(100))
hfile["group1/dataset2"] <- rgamma(10)
h5close(hfile)
```

So is reading data from files:

```r
hfile <- h5file("test.h5")
d1_sub <- hfile["group1/dataset1"][1:20]
names(d1_sub) <- h5attr(hfile["group1/dataset1"], "id")
```

For more examples, see the [package vignette](https://cran.r-project.org/web/packages/h5/vignettes/h5-Intro.html).

## Context

I think this package may be somewhat easier to use that `rhdf5`. It also has the advantage of being on CRAN (unlike `rhdf5`, which depends on Bioconductor). As far as I can tell, it also uses the system HDF5 library, which (a) makes it fit more naturally with the current system, and (b) significantly reduces installation time.

## Possible Implementation

I think most of our `rhdf5` use is in `PEcAn.ED2`, so that's the main place that would see any changes. But, given how easy this package makes it to work with HDF5 files, we could adapt them for temporary data storage in other contexts.

One limitation of this package is that it can't store mixed data types (e.g. `data.frame`s) as individual HDF5 datasets (though in fairness, I'm not sure if `rhdf5` can either, or if this is something we ever do.

**Comment by mdietze** [comment]
Sounds like a great idea to try implementing this in ED.

**Comment by dlebauer** [comment]
Does this also work with netcdf files?

**Comment by ashiklom** [comment]
@dlebauer Haven't tried. In theory, it should, since AFAIK NetCDF is HDF5 with specific rules and definitions. But I would be inclined to stick to NetCDF-specific packages, since modifying the underling HDF5 directly could potentially break the NetCDF standard and cause those files to become unreadable by software that expects properly formatted NetCDF.

**Comment by robkooper** [comment]
Here is a quick grep:

```
grep -R rhdf5 ~/git/pecan
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:## It requires the rhdf5 library, which is not available on CRAN, but by can be installed locally:
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:## >biocLite('rhdf5')
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:            rhdf5::h5createFile(mout)
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:          rhdf5::h5createFile(mout)
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(nbdsf, mout, "nbdsf")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(nddsf, mout, "nddsf")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(vbdsf, mout, "vbdsf")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(vddsf, mout, "vddsf")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(prate, mout, "prate")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(dlwrf, mout, "dlwrf")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(pres, mout, "pres")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(hgt, mout, "hgt")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(ugrd, mout, "ugrd")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(vgrd, mout, "vgrd")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(sh, mout, "sh")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:        rhdf5::h5write.default(tmp, mout, "tmp")
/Users/kooper/git/pecan/models/ed/R/met2model.ED2.R:          rhdf5::h5write.default(co2, mout, "co2")

/Users/kooper/git/pecan/models/ed/R/write_restart.ED2.R:      h5_write <- rhdf5::h5write.default(new.nplant_co_plant, histfile, "NPLANT")

/Users/kooper/git/pecan/models/ed/DESCRIPTION:    rhdf5
/Users/kooper/git/pecan/models/ed/DESCRIPTION:    github::bioconductor-mirror/rhdf5@release-3.4

/Users/kooper/git/pecan/scripts/install.dependencies.R:# install rhdf5 from bioconductor for met2model.ED
/Users/kooper/git/pecan/scripts/install.dependencies.R:if (!("rhdf5" %in% installed.packages()[, "Package"])) {
/Users/kooper/git/pecan/scripts/install.dependencies.R:  biocLite("rhdf5")
```

There is also some references in DART, but we should just mark that as depricated.

**Comment by dlebauer** [comment]
FYI I just found out over at https://github.com/rocker-org/geospatial/issues/10#issuecomment-359543125 that h5 is deprecated in favor of hdf5r: https://github.com/mannau/h5#deprecation-notice

**Comment by infotroph** [comment]
Closed in #1818
